egs/wsj/s5/steps/nnet3/train_discriminative.sh

```diff
@@ -0,0 +1,370 @@
+# using egs obtained by steps/nnet3/get_egs_discriminative.sh
+
+# Begin configuration section.
+cmd=run.pl
+num_epochs=4       # Number of epochs of training;
+                   # the number of iterations is worked out from this.
+                   # Be careful with this: we actually go over the data
+                   # num-epochs * frame-subsampling-factor times, due to
+                   # using different data-shifts.
+use_gpu=true
+truncate_deriv_weights=0  # can be used to set to zero the weights of derivs from frames
+                          # near the edges.  (counts subsampled frames).
+apply_deriv_weights=true
+run_diagnostics=true
+learning_rate=0.00002
+max_param_change=2.0
+scale_max_param_change=false # if this option is used, scale it by num-jobs.
+
+effective_lrate=    # If supplied, overrides the learning rate, which gets set to effective_lrate * num_jobs_nnet.
+acoustic_scale=0.1  # acoustic scale for MMI/MPFE/SMBR training.
+boost=0.0       # option relevant for MMI
+
+criterion=smbr
+drop_frames=false #  option relevant for MMI
+one_silence_class=true # option relevant for MPE/SMBR
+num_jobs_nnet=4    # Number of neural net jobs to run in parallel.  Note: this
+                   # will interact with the learning rates (if you decrease
+                   # this, you'll have to decrease the learning rate, and vice
+                   # versa).
+regularization_opts=
+minibatch_size=64  # This is the number of examples rather than the number of output frames.
+modify_learning_rates=false
+last_layer_factor=1.0  # relates to modify-learning-rates
+first_layer_factor=1.0 # relates to modify-learning-rates
+shuffle_buffer_size=1000 # This "buffer_size" variable controls randomization of the samples
+                # on each iter.  You could set it to 0 or to a large value for complete
+                # randomization, but this would both consume memory and cause spikes in
+                # disk I/O.  Smaller is easier on disk and memory but less random.  It's
+                # not a huge deal though, as samples are anyway randomized right at the start.
+
+
+stage=-3
+
+adjust_priors=true
+num_threads=16  # this is the default but you may want to change it, e.g. to 1 if
+                # using GPUs.
+
+cleanup=true
+keep_model_iters=1
+retroactive=false
+remove_egs=false
+src_model=  # will default to $degs_dir/final.mdl
+
+left_deriv_truncate=   # number of time-steps to avoid using the deriv of, on the left.
+right_deriv_truncate=  # number of time-steps to avoid using the deriv of, on the right.
+# End configuration section.
+
+
+echo "$0 $@"  # Print the command line for logging
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. parse_options.sh || exit 1;
+
+
+if [ $# != 2 ]; then
+  echo "Usage: $0 [opts] <degs-dir> <exp-dir>"
+  echo " e.g.: $0 exp/tri4_mpe_degs exp/tri4_mpe"
+  echo ""
+  echo "You have to first call get_egs_discriminative2.sh to dump the egs."
```

Comment:
remove all references to  get_egs_discriminative2.sh- search in the PR.  [if I merge anyway, do it as a separtate PR].